### PR TITLE
Provide basic unit tests for FDT interface

### PR DESCRIPTION
**Pull request by Tomatosoup97**
Despite of the fact that the tests will assure that operations on FDT work as expected, they will provide us a basic usage guide.

### CODE REVIEW

**Comment by mzr** [comment]
Kernel tests are failing. What's te state of this PR ?

File: tests/fdt.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  assert(fdt_path_offset(fdt, "/pci@0/isa@0") != -1);
+  assert(fdt_path_offset(fdt, "/pci@0/isa@0/rtc@0") != -1);
+  assert(fdt_path_offset(fdt, "/pci/isa") != -1);
+  assert(fdt_path_offset(fdt, "/pci/isa/rtc@0") != -1);
+  assert(fdt_path_offset(fdt, "/non/existing/path") == -1);
+  return KTEST_SUCCESS;
+}
+
+static int test_fdt_get_path(void) {
+  const char *expected_path = "/pci@0/isa@/rtc@0";
```
Comment (mzr) [inline tests/fdt.c]: Typo? `isa@` -> `isa@0` ?

Comment (Tomatosoup97) [reply tests/fdt.c]: Fixed

**Comment by cahirwpz** [comment]
This PR seems to be useful, are there any chances it will get prepared for merge soon?

**Comment by cahirwpz** [comment]
This PR isn't that valuable anymore. I decided to close it.

**Pull request closed without merging**